src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -2127,7 +2127,229 @@ private function step_in_body(): bool {
 	 * @return bool Whether an element was found.
 	 */
 	private function step_in_table(): bool {
-		$this->bail( 'No support for parsing in the ' . WP_HTML_Processor_State::INSERTION_MODE_IN_TABLE . ' state.' );
+		$token_name = $this->get_token_name();
+		$token_type = $this->get_token_type();
+		$op_sigil   = '#tag' === $token_type ? ( parent::is_tag_closer() ? '-' : '+' ) : '';
+		$op         = "{$op_sigil}{$token_name}";
+
+		switch ( $op ) {
+			/*
+			 * > A character token, if the current node is table, tbody, template, tfoot, thead, or tr element
+			 */
+			case '#text':
+				if (
+					$this->state->stack_of_open_elements->current_node_is( 'TABLE' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'TBODY' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'TEMPLATE' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'TFOOT' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'THEAD' ) ||
+					$this->state->stack_of_open_elements->current_node_is( 'TR' )
```

Comment:
maybe this helps, but in the case of comparing against multiple items I don't find the call to `current_node_is()` more helpful can capturing the current node name and comparing to the names.

Reply:
I'll apply this suggestion, I don't feel strongly either way.